lib/maths/unittest/CBoostedTreeTest.cc

```diff
@@ -40,13 +41,52 @@
 #include <utility>
 #include <vector>
 
-BOOST_AUTO_TEST_SUITE(CBoostedTreeTest)
+namespace ml {
+namespace maths {
+class CBoostedTreeImplForTest {
+public:
+    using TAnalysisInstrumentationPtr = CBoostedTreeImpl::TAnalysisInstrumentationPtr;
+    using TLossFunctionUPtr = CBoostedTreeImpl::TLossFunctionUPtr;
+    using TSizeVec = CBoostedTreeImpl::TSizeVec;
+    using TDoubleVec = CBoostedTreeImpl::TDoubleVec;
+
+public:
+    CBoostedTreeImplForTest(CBoostedTreeImpl& treeImpl)
```

Comment:
nit: maybe mark the single-argument constructor as explicit?

Reply:
[88f9b71](https://github.com/elastic/ml-cpp/pull/1741/commits/88f9b71ebe2a8d2a2956f49e76c24fdf449ac34b)